src/entities/activity/model/mappers.ts

```diff
@@ -64,13 +63,82 @@ function mapConditionalLogic(dto: ConditionalLogicDto | null) {
         conditionalLogic: {
           match: dto.match,
           conditions: dto.conditions.map(condition => {
+            const { itemName, ...rest } = condition;
             const updatedCondition = {
-              ...condition,
-              activityItemName: condition.itemName,
+              ...rest,
+              activityItemName: itemName,
             };
-
             // @ts-expect-error
             delete updatedCondition.itemName;
+            switch (condition.type) {
+              case 'INCLUDES_OPTION':
+              case 'NOT_INCLUDES_OPTION':
+              case 'EQUAL_TO_OPTION':
+              case 'NOT_EQUAL_TO_OPTION':
+                updatedCondition.payload = {
+                  optionValue: condition.payload.optionValue,
+                };
+                break;
+
+              case 'INCLUDES_ROW_OPTION':
+              case 'NOT_INCLUDES_ROW_OPTION':
+                updatedCondition.payload = {
+                  optionValue: condition.payload.optionValue,
+                  rowIndex: condition.payload.rowIndex,
+                };
+                break;
+
+              case 'GREATER_THAN_DATE':
+              case 'LESS_THAN_DATE':
+              case 'EQUAL_TO_DATE':
+              case 'NOT_EQUAL_TO_DATE':
+                updatedCondition.payload = { date: condition.payload.date };
+                break;
+
+              case 'BETWEEN_DATES':
+              case 'OUTSIDE_OF_DATES':
```

Comment:
The cases `'BETWEEN_TIMES'`, `'OUTSIDE_OF_TIMES'`, `'BETWEEN_TIME_RANGES'` and `'OUTSIDE_OF_TIME_RANGES'` can be grouped with these two, all are passing both `minDate` and `maxDate`.

This could be left as is to preserve ordering. Order for this switch can be replicated from the actual `Condition` type